vignettes/backtesting.Rmd

```diff
@@ -84,384 +74,363 @@ doctor_visits <- pub_covidcast(
 implementation passes a simple validation.
 
-Now let's go ahead and slide this forecaster in a version unaware way and a
-version aware way. For the version unaware way, we need to snapshot the latest
-version of the data, and then make a faux archive by setting `version =
-time_value`. This has the effect of simulating a data set that receives the
-final version updates every day. For the version aware way, we will simply use
-the true `epi_archive` object.
+## Comparing version faithful and version un-faithful forecasts
+
+Now let's compare the behavior of this forecaster, both properly considering data versioning 
+("version faithful") and ignoring data versions ("version un-faithful").
+
+For the version un-faithful approach, we need to do some setup if we want to use `epix_slide` for backtesting.
+We want to simulate a data set that receives finalized updates every day, that is, a data set with no revisions.
+To do this, we will snapshot the latest version of the data to create a synthetic data set, and convert it into an archive
+where `version = time_value`[^2].
 
 ```{r}
-archive_cases_dv_subset_faux <- doctor_visits %>%
-  epix_as_of(doctor_visits$versions_end) %>%
-  mutate(version = time_value) %>%
+archive_cases_dv_subset_faux <- doctor_visits |>
+  epix_as_of(doctor_visits$versions_end) |>
+  mutate(version = time_value) |>
   as_epi_archive()
 ```
 
-To reduce typing, we create the wrapper function `forecast_k_week_ahead()`.
+For the version faithful approach, we will continue using the original `epi_archive` object containing all version updates.
+
+We will also create the helper function `forecast_wrapper()` to let us easily map across aheads.
 
 ```{r arx-kweek-preliminaries, warning = FALSE}
-# Latest snapshot of data, and forecast dates
-forecast_dates <- seq(from = as.Date("2020-08-01"), to = as.Date("2021-11-01"), by = "1 month")
-aheads <- c(7, 14, 21, 28)
-
-# @param epi_archive The epi_archive object to forecast from
-# @param ahead The number of days ahead to forecast
-# @param outcome The outcome variable to forecast
-# @param predictors The predictors to use in the model
-# @param forecast_dates The dates to forecast on
-# @param process_data A function to process the data before forecasting
-forecast_k_week_ahead <- function(
-    epi_archive,
-    ahead = 7,
-    outcome = NULL, predictors = NULL, forecast_dates = NULL, process_data = identity) {
-  if (is.null(forecast_dates)) {
-    forecast_dates <- epi_archive$versions_end
-  }
-  if (is.null(outcome) || is.null(predictors)) {
-    stop("Please specify the outcome and predictors.")
-  }
-  epi_archive %>%
-    epix_slide(
-      ~ arx_forecaster(
-        process_data(.x), outcome, predictors,
-        args_list = arx_args_list(ahead = ahead)
-      )$predictions %>%
-        pivot_quantiles_wider(.pred_distn),
-      .before = 120,
-      .versions = forecast_dates
-    )
+forecast_wrapper <- function(
```

Comment:
Can we format this chunk a bit better: there's lots of available width. It just
kind of looks funny.

Reply:
mostly trying to rely on tabs to indicate sub-calls, since it's pretty deeply nested